cmd/cli/root.go

```diff
@@ -0,0 +1,94 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	mcobra "github.com/muesli/mango-cobra"
+	"github.com/muesli/roff"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+
+	"github.com/bom-squad/go-cli/cmd/cli/options"
+	"github.com/bom-squad/go-cli/pkg/log"
+)
+
+var version = "dev"
+
+func ManCommand(root *cobra.Command) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "man",
+		Short:                 "Generates command line manpages",
+		SilenceUsage:          true,
+		DisableFlagsInUseLine: true,
+		Hidden:                true,
+		Args:                  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			manPage, err := mcobra.NewManPage(1, root)
+			if err != nil {
+				return err
+			}
+
+			_, err = fmt.Fprint(os.Stdout, manPage.Build(roff.NewDocument()))
+			return err
+		},
+	}
+
+	return cmd
+}
+
+func NewRootCmd() *cobra.Command {
+	ro := &options.RootOptions{}
+	rootCmd := &cobra.Command{
+		Use:     "sbom-convert",
+		Version: version,
+		Short:   "",
+		Long:    ``,
+		Run:     func(cmd *cobra.Command, args []string) {},
+		CompletionOptions: cobra.CompletionOptions{
+			HiddenDefaultCmd: true,
+		},
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := validateRootOptions(ro); err != nil {
+				return err
+			}
+			return setupLogger(ro)
+		},
+		SilenceErrors: true,
+	}
+
+	ro.AddFlags(rootCmd)
+
+	// Commands
+	rootCmd.AddCommand(ConvertCommand())
+
+	// Manpages
+	rootCmd.AddCommand(ManCommand(rootCmd))
+	return rootCmd
+}
+
+func validateRootOptions(_ *options.RootOptions) error {
+	return nil
+}
+
+func setupLogger(ro *options.RootOptions) error {
```

Comment:
i suggest we somehow allow the CLI to pass a instance of the logger to the library or some way for the logger to access the global logger, 

I was thinking we should use a interface so the Library can add all  shorts of information to the logger.
but they may be other methods to do this, or simply decide to chose a specific logger to use across the repos.

Reply:
I have raised a concern regarding `protobom` as it should only function as a library without directly requiring a logger or outputting anything to stdout unless specified. However, I am not in favor of including a logger with stdout. 

I concur that it would be best to support an external logger (passed from the CLI) as an interface that defaults to stdout/log.

Reply:
>  I am not in favor of including a logger with stdout.
Can you elaberate what you mean i don't fully follow.

I guess we can leave the logger out of this PR, lets try and discuss this @puerco @manifestori See if we can present a interface (we need both the cli and library to agree on this).